packages/react-async/src/useAsync.tsx

```diff
@@ -33,27 +32,16 @@ declare type ImportWorkaround<T> =
   | AsyncPending<T>
   | AsyncRejected<T>
 
-export interface FetchOptions<T> extends AsyncOptions<T> {
+export interface FetchOptions<T, C> extends AsyncOptions<T, C> {
   defer?: boolean
   json?: boolean
 }
 
-function useAsync<T>(options: AsyncOptions<T>): AsyncState<T>
-function useAsync<T>(promiseFn: PromiseFn<T>, options?: AsyncOptions<T>): AsyncState<T>
-
-function useAsync<T>(arg1: AsyncOptions<T> | PromiseFn<T>, arg2?: AsyncOptions<T>): AsyncState<T> {
-  const options: AsyncOptions<T> =
-    typeof arg1 === "function"
-      ? {
-          ...arg2,
-          promiseFn: arg1,
-        }
-      : arg1
-
+function useAsync<T, C>(options: AsyncOptions<T, C>): AsyncState<T> {
   const counter = useRef(0)
   const isMounted = useRef(true)
-  const lastArgs = useRef<any[] | undefined>(undefined)
-  const lastOptions = useRef<AsyncOptions<T>>(options)
+  const lastArgs = useRef<C | undefined>(undefined)
```

Comment:
Shouldn't this just be:

```suggestion
  const lastArgs = useRef<C>(undefined)
```

I haven't tested, but I think this may remove the need to explicitly cast to `C` in a few places.